testing/test_postprocessing.py

```diff
@@ -64,45 +64,54 @@ def test_threshold(nii_seg):
     assert np.array_equal(nii_seg_proc.get_fdata()[4:7, 8, 4], np.array([0, 1, 1]))
 
 
-@pytest.mark.parametrize('nii_seg', [nii_dummy_seg()])
+@pytest.mark.parametrize('nii_seg', [nii_dummy_seg(), nii_dummy_seg(softseg=True)])
 def test_keep_largest_object(nii_seg):
+    nii_seg_copy = np.copy(nii_seg)
     # Set a voxel to 1 at the corner to make sure it is set to 0 by the function
     coord = (1, 1, 1)
     nii_seg.dataobj[coord] = 1
     # Test function with array input
     arr_seg_proc = postproc.keep_largest_object(np.copy(np.asanyarray(nii_seg.dataobj)))
     assert isinstance(arr_seg_proc, np.ndarray)
     assert arr_seg_proc[coord] == 0
+    # Make sure it equals the input data, in particular: still binary / soft if the input was binary / soft
+    assert np.array_equal(nii_seg.dataobj, arr_seg_proc)
     # Make sure it works with nibabel input
     nii_seg_proc = postproc.keep_largest_object(nii_seg)
     assert isinstance(nii_seg_proc, nib.nifti1.Nifti1Image)
     assert nii_seg_proc.dataobj[coord] == 0
 
 
-@pytest.mark.parametrize('nii_seg', [nii_dummy_seg()])
+@pytest.mark.parametrize('nii_seg', [nii_dummy_seg(), nii_dummy_seg(softseg=True)])
 def test_keep_largest_object_per_slice(nii_seg):
+    nii_seg_copy = np.copy(nii_seg)
     # Set a voxel to 1 at the corner to make sure it is set to 0 by the function
     coord = (1, 1, 1)
     nii_seg.dataobj[coord] = 1
     # Test function with array input
     arr_seg_proc = postproc.keep_largest_object_per_slice(np.copy(np.asanyarray(nii_seg.dataobj)), axis=2)
     assert isinstance(arr_seg_proc, np.ndarray)
     assert arr_seg_proc[coord] == 0
+    # Make sure it equals the input data, in particular: still binary / soft if the input was binary / soft
+    assert np.array_equal(nii_seg.dataobj, arr_seg_proc)
```

Comment:
not sure i understand this assert: the purpose of preprocessing is to *modify* the data, so we should not expect the input to equal the output. Or am i missing something?  Maybe you wanted to test if a soft input produces a a soft output (ditto for binary input/output)

Reply:
My understanding of these test functions is:
1. Alter the input dummy seg on a particular coordinate
2. Run postpro
3. Check if postprocessing did want we expect from it at this particular coordinate
--> I would like, with this test I added, to check if the input (ie before alteration) is not modified.